crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -64,6 +64,24 @@ def services
     end
   end
 
+  api :POST, "/api/upgrade/controllers", "Orchestrate the upgrade of controller nodes"
```

Comment:
For the record, this discussion was resolved elsewhere. We want to use only one endpoint (`/nodes`), run the method asynchronously, and use some global status information to track current state.